closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -50,6 +50,10 @@
 #define OPENSSL_VERSION_3_0_0 OPENSSL_VERSION_CODE(3, 0, 0, 0)
 #define OPENSSL_VERSION_4_0_0 OPENSSL_VERSION_CODE(4, 0, 0, 0)
 
+/* Defining OpenSSL operation modes. */
+#define OPENSSL_ENCRYPTION_MODE 1
+#define OPENSSL_DECRYPTION_MODE 0
```

Comment:
I don't think "Defining" is helpful here, but you might make reference to `OSSL_CipherInit_ex()` where the constants are used. I also think you should add and use something like
```c
#define OPENSSL_SAME_MODE (-1)
```
Please use these names instead of integer literals where appropriate (e.g. line 1304 and in `ChaCha20Init()`).

Reply:
Done.